src/Symfony/Component/DependencyInjection/Container.php

```diff
@@ -419,6 +419,14 @@ protected function getEnv($name)
         if (false !== $env = getenv($name)) {
             return $this->envCache[$name] = $env;
         }
+        if (0 < $i = strpos($name, ':')) {
+            $id = substr($name, 0, $i);
+            $service = isset($this->services[$id]) ? $this->services[$id] : (isset($this->methodMap[$id]) ? $this->{$this->methodMap[$id]}() : $this->get($id));
+
+            if (null !== $env = $service->getEnv(substr($name, 1 + $i))) {
```

Comment:
What about `processEnv($name, $value)` then? And continue with the `%env:service(key)%` syntax (or square brackets, i dont mind). We should avoid the term _lookup_.

Besides you can still process `$value` based on `$key` if needed (ie. do a lookup). But i think simply decorating `$value` is most common.